tests/integration/test_logcollector/test_log_filter_options/test_restrict_ignore_regex.py

```diff
@@ -0,0 +1,185 @@
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set the wazuh configuration according to the configuration data.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Configure the local_internal_options file.
+        - restart_wazuh_function:
+            type: fixture
+            brief: Restart wazuh.
+
+    assertions:
+        - Check that logcollector is analyzing the log file.
+        - Check that logs are ignored when they do not match with configured regex
+
+    input_description:
+        - The `configuration_restrict_ignore_regex_values.yaml` file provides the module configuration for this test.
+        - The `cases_restrict_ignore_regex_values.yaml` file provides the test cases.
+
+    expected_output:
+        - r".*wazuh-logcollector.*Analizing file: '{file}'.*"
+        - r".*wazuh-logcollector.*DEBUG: Reading syslog '{message}'.*"
+        - r".*wazuh-logcollector.*DEBUG: Ignoring the log line '{message}' due to {tag} config: '{regex}'"
+    '''
+    log = metadata['log_sample']
+    command = f"echo {log}>> {test_file}"
+
+    if sys.platform == 'win32':
+        file = re.escape(test_file)
+    else:
+        file = test_file
+
+    # Check log file is being analized
+    evm.check_analyzing_file(file=file, prefix=prefix)
+
+    # Insert log
+    run_local_command_returning_output(command)
+    # Check the log is read from the monitored file
+    evm.check_syslog_messages(message=log, prefix=prefix)
+
+    # Check responses
+    # If it matches with ignore, it should ignore the log due to ignore config
+    if 'ignore' in metadata['matches']:
+        evm.check_ignore_restrict_messages(message=log, regex=metadata['ignore_regex'], tag='ignore',
+                                           prefix=prefix)
+        if 'restrict' in metadata['matches']:
+            evm.check_ignore_restrict_message_not_found(message=log, regex=metadata['restrict_regex'], tag='restrict',
+                                                        prefix=prefix)
+
+    # If matches with restrict, it should not be ignored due to restrict config
+    elif metadata['matches'] == 'restrict':
+        evm.check_ignore_restrict_message_not_found(message=log, regex=metadata['restrict_regex'], tag='restrict',
+                                                    prefix=prefix)
+        evm.check_ignore_restrict_message_not_found(message=log, regex=metadata['ignore_regex'], tag='ignore',
+                                                    prefix=prefix)
+
+    # If it matches with None, the log should be ignored due to restrict config and not due to ignore config
+    else:
+        evm.check_ignore_restrict_message_not_found(message=log, regex=metadata['ignore_regex'], tag='ignore',
+                                                    prefix=prefix)
+        evm.check_ignore_restrict_messages(message=log, regex=metadata['restrict_regex'], tag='restrict',
```

Comment:
No, because, which tag? With that name one coud assume it could work for any tags, but it is only for ignore/restrict messages, it uses the tag as parameter, but it does not only check the tag.